packages/webapp/src/util/task.ts

```diff
@@ -172,6 +172,24 @@ export const formatSoilAmendmentProductToDBStructure = (
   });
 };
 
+export const formatSoilAmendmentTaskToDBStructure = (
+  soilAmendmentTask: FormSoilAmendmentTask,
+  { methods }: { methods: SoilAmendmentMethod[] },
+): DBSoilAmendmentTask => {
+  const soilAmendmentTaskClone = structuredClone(soilAmendmentTask);
+  const furrowHoleId = methods?.find(({ key }) => key === 'FURROW_HOLE')?.id;
+  if (!furrowHoleId) {
+    throw Error('id for FURROW_HOLE method does not exist');
+  }
+  soilAmendmentTaskClone.furrow_hole_depth_unit =
+    soilAmendmentTaskClone.furrow_hole_depth_unit.value;
```

Comment:
- This needs the condition: `if (soilAmendmentTask.method_id === furrowHoleId)`
- Would you mind deleting `other_application_method` when "Other" is not selected as well?

Reply:
- I think it is always present as the only unit option {"label": "cm", "value": "cm"} -- and the opposite condition is below if not needed. But I can switch to the positive condition.